Source/WebKit/NetworkProcess/cache/NetworkCache.cpp

```diff
@@ -715,5 +716,67 @@ void Cache::storeData(const DataKey& dataKey, const uint8_t* data, size_t size)
     m_storage->store(record, { });
 }
 
+void Cache::fetchData(bool shouldComputeSize, CompletionHandler<void(Vector<WebsiteData::Entry>&&)>&& completionHandler)
+{
+    HashMap<WebCore::SecurityOriginData, uint64_t> originsAndSizes;
+    traverse([protectedThis = Ref { *this }, shouldComputeSize, completionHandler = WTFMove(completionHandler), originsAndSizes = WTFMove(originsAndSizes)](auto* traversalEntry) mutable {
```

Comment:
We should probably make a Cache::traverse that doesn't have the check for maximumTraverseCount for things like this.  For other uses it's ok to pretend we don't have a cache if there's too much congestion, but for fetching and deleting data we need to be sure we got it all.  Can be done in a follow-up because that is not changed in this PR.

Reply:
Sounds good; I will file a bug for that.